test-control/src/main/java/com/pkb/common/testcontrol/services/DefaultProcessingControllerService.java

```diff
@@ -1,43 +1,52 @@
 package com.pkb.common.testcontrol.services;
 
 import com.pkb.common.testcontrol.camel.route.AbstractTestControlCamelRouteBuilder;
+import com.pkb.common.testcontrol.message.ResumeProcessingRequest;
+import com.pkb.common.testcontrol.message.SuspendProcessingRequest;
 import org.apache.camel.CamelContext;
 import org.apache.camel.Route;
 import org.apache.camel.impl.engine.AbstractCamelContext;
 import org.apache.camel.spi.RouteStartupOrder;
 import org.apache.camel.util.function.ThrowingConsumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.Comparator;
 
 import static com.pkb.common.testcontrol.client.TestControl.IO_PKB_TESTCONTROL_PREFIX;
 
-public class DefaultPubSubNamespaceService implements PubSubNamespaceService {
+public class DefaultProcessingControllerService implements ProcessingControllerService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(java.lang.invoke.MethodHandles.lookup().lookupClass());
+
     private final CamelContext context;
-    
-    private String currentNamespace = "defaultNS";
 
-    public DefaultPubSubNamespaceService(CamelContext context) {
+    public DefaultProcessingControllerService(CamelContext context) {
         this.context = context;
     }
 
     @Override
-    public void setCurrentNamespace(String currentNamespace) {
-        // We want to ensure that all _currently_ processing messages are stopped before we change the namespace.
-        // This ensures that any unprocessed messages are ignored and don't cause havoc by running while we try to 
-        // reset the application internal state.
+    public void process(SuspendProcessingRequest message) {
+        LOGGER.info("SuspendProcessingRequest.process message received");
         applyToRoutesExceptTestControl(route -> context.getRouteController().suspendRoute(route.getRouteId()), true);
-        this.currentNamespace = currentNamespace;
+        LOGGER.info("SuspendProcessingRequest.process done.");
+    }
+
+    @Override
+    public void process(ResumeProcessingRequest message) {
+        LOGGER.info("ResumeProcessingRequest.process message received");
         applyToRoutesExceptTestControl(route -> context.getRouteController().resumeRoute(route.getRouteId()), false);
+        LOGGER.info("ResumeProcessingRequest.process done.");
     }
 
     private void applyToRoutesExceptTestControl(ThrowingConsumer<Route, Exception> routeConsumer, boolean reverseOrder) {
-        // This reproduces a bit of logic inside camel; but adding filtering so we can startup and shutdown 
+        // This reproduces a bit of logic inside camel; but adding filtering so we can startup and shutdown
```

Comment:
Does test-control have camel routes? I thought it used http messages now

Reply:
Yes, those http endpoints are set up via camel RestDefinitions, and we don't want to disable them. See here: https://github.com/patientsknowbest/pkb-common/blob/1e3b6b22a40fb65c5ff074abbdf075bed5548162/test-control/src/main/java/com/pkb/common/testcontrol/camel/route/AbstractTestControlCamelRouteBuilder.java#L69